packages/server/src/queries/complex/pools/transmuter.ts

```diff
@@ -0,0 +1,66 @@
+import { CoinPretty, Dec, RatePretty } from "@keplr-wallet/unit";
+import { AssetList, Chain } from "@osmosis-labs/types";
+import cachified, { CacheEntry } from "cachified";
+import { LRUCache } from "lru-cache";
+
+import { captureIfError, DEFAULT_LRU_OPTIONS } from "../../../utils";
+import { queryTransmuterTotalPoolLiquidity } from "../../contracts";
+import { Asset, getAsset } from "../assets";
+
+const transmuterTotalPoolLiquidityCache = new LRUCache<string, CacheEntry>(
+  DEFAULT_LRU_OPTIONS
+);
+
+export async function getCachedTransmuterTotalPoolLiquidity(
+  contractAddress: string,
+  chainList: Chain[],
+  assetLists: AssetList[]
+) {
+  return await cachified({
+    cache: transmuterTotalPoolLiquidityCache,
+    ttl: 1000 * 30, // 30 seconds
+    key: `transmuter-total-pool-liquidity-${contractAddress}`,
+    getFreshValue: async (): Promise<
+      Array<{
+        asset: Asset;
+        coin: CoinPretty;
+        percentage: RatePretty;
+      }>
+    > => {
+      const poolLiquidityAssets: Array<{
+        asset: Asset;
+        coin: CoinPretty;
+        percentage: RatePretty;
+      }> = [];
+
+      const {
+        data: { total_pool_liquidity = [] },
+      } = await queryTransmuterTotalPoolLiquidity({
+        contractAddress,
+        chainList,
+      });
+
+      const totalLiquidity = total_pool_liquidity.reduce((acc, asset) => {
+        return acc.add(new Dec(parseInt(asset.amount, 10)));
```

Comment:
Is this `parseInt` needed? Dec accepts strings, and using JS numbers could result in precision/MAXINT issues

Reply:
Not required, if Dec support strings I'll just remove `parseInt`, good suggestion! Thank you

Reply:
done